custom_typings/estree.d.ts

```diff
@@ -1,12 +1,17 @@
 // TODO(rictic): Upstream/merge with https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/estree/estree.d.ts
 
 declare module 'estree' {
-  export interface Node {
-    type: string;
+  interface BaseNode {
     leadingComments?: Comment[];
```

Comment:
That works, but it would paper over mistakes in BaseNode leaf interfaces. It's better that it be missing so that the type of `Node.type` is always a union of string literal types.

Added a comment describing this.